### PR TITLE
Documenting testing framework

### PR DESCRIPTION
**Pull request by alex-konovalov**
**Related issues**

Refs: #165 

**Describe the changes made in this pull request**

Described how the tests are organised and how to run them locally.

**Instructions to review the pull request**

@jspaaks this is my understanding of how this works by looking at the current state of the testing code - I didn't look at the history and I may be wrong. Also, formulations may not follow the style used in other parts of instructions - I will be happy to correct and force-push my branch following the suggestions.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -50,6 +50,17 @@ Please **fork the repository** you want to contribute to, and then
...
 
+There is a collection of test `.cff` files in the `test` directory.
+It may be a good idea to add a test if you modify the specification,
+or if you discover a part of the specification which is not covered
+by tests. If you modify anything in the `test` directory, it is
+advised to run these tests locally on your computer prior to submitting
+a pull request. However, if that's not possible, you still can submit
+the pull request and later check the status of the tests for your
+pull requests on GitHub. Please see `test/README.md` file for further
+details about tests organisations and instructions how to run them
```
Comment (jspaaks) [inline CONTRIBUTING.md:61]:
```suggestion
details about tests and instructions how to run them
```

File: test/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+## Dependencies
+
+- Python 3.6 or higher
+- [cffconvert](https://pypi.org/project/cffconvert/) version 1.0.0 or higher
+- [pytest](https://pypi.org/project/pytest/)
+
+You can install `cffconvert` and `pytest` using the `requirements.txt` file from
+this directory by entering the following command:
+
+    python3 -m pip install -r requirements.txt
```
Comment (jspaaks) [inline test/README.md:20]:
```suggestion
    python3 -m pip install -r test/requirements.txt
```
(just to clarify which `requirements.txt` we mean, the other one from `/` may disappear later)

**Review by jspaaks — CHANGES_REQUESTED** [review]
Nicely done! I left some minor suggestions

**Comment by alex-konovalov** [comment]
Thanks @jspaaks - updated following your suggestions.

**Review by jspaaks — APPROVED** [review]
Good stuff, thanks for making this

**Pull request merged**